src/Centreon/Application/Controller/Monitoring/CommentController.php

```diff
@@ -0,0 +1,346 @@
+            $comments[$index] = (new Comment($commentResource['id'], $commentResource['comment']))
+                ->setDate($date)
+                ->setParentResourceId($commentResource['parent']['id']);
+
+            if ($commentResource['type'] === ResourceEntity::TYPE_HOST) {
+                $hostIds[$index] = $commentResource['id'];
+            } elseif ($commentResource['type'] === ResourceEntity::TYPE_SERVICE) {
+                $serviceIds[$index] = [
+                    'host_id' => $commentResource['parent']['id'],
+                    'service_id' => $commentResource['id']
+                ];
+            }
+        }
+
+        /**
+         * Retrieving all services and hosts
+         */
+        $hosts = $this->monitoringService->findMultipleHosts($hostIds);
+        $services = $this->monitoringService->findMultipleServices($serviceIds);
+
+        if (!empty($hosts)) {
+            foreach ($hosts as $key => $host) {
+                $this->commentService->addHostComment($comments[$key], $host);
+            }
+        }
+
+        if (!empty($services)) {
+            foreach ($services as $key => $service) {
+                $this->commentService->addServiceComment($comments[$key], $service);
+            }
+        }
+
+        return $this->view(null, Response::HTTP_NO_CONTENT);
+    }
+
+    /**
+     * Entry point to add a comment on a host resource
+     *
+     * @param Request $request
+     * @param int $hostId ID of the host
+     * @return View
+     * @throws \Exception
+     * @throws \InvalidArgumentException
+     */
+    public function addHostComment(
+        Request $request,
+        int $hostId
+    ): View {
+        $this->denyAccessUnlessGrantedForApiRealtime();
+        /**
+         * @var Contact $contact
+         */
+        $contact = $this->getUser();
+
+        /**
+         * Checking that user is allowed to add a comment for a host resource
+         */
+        if (!$contact->isAdmin() && !$contact->hasRole(Contact::ROLE_HOST_ADD_COMMENT)) {
+            return $this->view(null, Response::HTTP_UNAUTHORIZED);
+        }
+
+        $results = $this->validateAndRetrievePostData(
+            $request,
+            'config/json_validator/latest/Centreon/Comment/Comment.json'
+        );
+
+        if (!empty($results)) {
+            /**
+             * At this point we made sure that the mapping will work since we validate
+             * the JSON sent with the JSON validator.
+             */
+            $host = $this->monitoringService->findOneHost($hostId);
+            if (is_null($host)) {
+                throw new EntityNotFoundException(
+                    sprintf(
+                        _('Host %d not found'),
+                        $hostId
+                    )
+                );
+            }
+            $date = ($results['date'] !== null) ? new \DateTime($results['date']) : new \DateTime('now');
```

Comment:
```suggestion
            $date = ($results['date'] !== null) ? new \DateTime($results['date']) : new \DateTime();
```
'now' is the default value for a new \DateTime()